api/src/opentrons/protocol_reader/file_reader_writer.py

```diff
@@ -63,7 +68,24 @@ async def _read_file(
 
             if filename.lower().endswith(".json"):
                 try:
-                    data = parse_raw_as(BufferedJsonFileData, contents)  # type: ignore[arg-type]
+                    json_dict = json.loads(contents)
+                    if (
+                        json_dict.get("ordering") is not None
+                        and json_dict.get("wells") is not None
+                    ):
+                        data = parse_raw_as(LabwareDefinition, contents)
+                    elif (
+                        json_dict.get("schemaVersion")
+                        and json_dict.get("commands") is not None
+                    ):
+                        if json_dict.get("schemaVersion") == 6:
+                            data = parse_raw_as(ProtocolSchemaV6, contents)
+                        else:
+                            data = parse_raw_as(JsonProtocol, contents)
+                    else:
+                        raise UnknownJsonFileError(
+                            f"JSON file {filename} has unknown Opentrons format."
```

Comment:
Slight message tweak

```suggestion
                            f"JSON file {filename} is not a known Opentrons format."
```

I also think the error message below from a `ValidationError` needs to be changed to reflect the fact that the JSON file _looked_ like a valid format, but failed validation. Something like:

```python
raise FileReadError(
    f"JSON file {filename} has missing or incorrect data."
) from e
```